gui/app/reducers/related_holdings.js

```diff
@@ -0,0 +1,56 @@
+import {
+    PULL_RELATED_HOLDINGS,
+    PULL_RELATED_HOLDINGS_SUCCESS,
+    PULL_RELATED_HOLDINGS_FAILED} from '../actions/related_holdings';
+import { SELECT_BIB_RECORD } from '../actions/global';
+import update from 'immutability-helper';
+
+export const produceInitialState = () => ({
+    loading: false,
+    relatedHoldings: [],
+    errorMessage: '',
+    selectedBibRecordId: null,
+    selectedBibAgencyId: null
+});
+
+export default function relatedHoldings(state = produceInitialState(), action = {}) {
+    switch (action.type){
+        case SELECT_BIB_RECORD:
+            // handling invalid actions
+            let validAction = !!(action.item) &&
+                (action.item.bibliographicRecordId !== undefined) &&
+                (action.item.agencyId !== undefined);
+            action.item = (action.item) ? action.item : {
+                bibliographicRecordId: null,
+                agencyId: null
+            };
+            let actionBibId = action.item.bibliographicRecordId;
+            let actionAgencyId = action.item.agencyId;
+            // If values are undefined, set to null instead
```

Comment:
Implementeret input validering i reducer, ville også gøre det for de andre reducere i det store test PR med UI integration tests og forbedringer til de eksisterende ruder tests.